tests/e2e/multi_vc_preferential_topology.go

```diff
@@ -0,0 +1,776 @@
+		}()
+
+		ginkgo.By("Create 3 statefulset with 10 replicas")
+		statefulSets := createParallelStatefulSetSpec(namespace, sts_count, stsReplicas)
+		var wg sync.WaitGroup
+		wg.Add(sts_count)
+		for i := 0; i < len(statefulSets); i++ {
+			go createParallelStatefulSets(client, namespace, statefulSets[i],
+				stsReplicas, &wg)
+
+		}
+		wg.Wait()
+
+		ginkgo.By("Waiting for StatefulSets Pods to be in Ready State")
+		err = waitForStsPodsToBeInReadyRunningState(ctx, client, namespace, statefulSets)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			deleteAllStsAndPodsPVCsInNamespace(ctx, client, namespace)
+		}()
+
+		ginkgo.By("Verify PV node affinity and that the PODS are running on appropriate node")
+		for i := 0; i < len(statefulSets); i++ {
+			verifyPVnodeAffinityAndPODnodedetailsForStatefulsetsLevel5(ctx, client, statefulSets[i],
+				namespace, allowedTopologies, parallelStatefulSetCreation, true)
+		}
+
+		ginkgo.By("Verify volume is provisioned on the preferred datatsore")
+		for i := 0; i < len(statefulSets); i++ {
+			err = verifyVolumeProvisioningForStatefulSet(ctx, client, statefulSets[i], namespace,
+				preferredDatastorePaths, nil, true, true, true, dsUrls)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Remove preferred datatsore tag which is chosen for volume provisioning")
+		for i := 0; i < len(preferredDatastorePaths); i++ {
+			err = detachTagCreatedOnPreferredDatastore(masterIp, sshClientConfig, preferredDatastorePaths[i],
+				allowedTopologies[0].Values[i], true, i)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		var preferredDatastorePathsNew []string
+		ginkgo.By("Tag new preferred datastore for volume provisioning in VC1 and VC2")
+		for i := 0; i < 2; i++ {
+			paths, err := tagPreferredDatastore(masterIp, sshClientConfig, allowedTopologies[0].Values[i],
+				preferredDatastoreChosen, ClusterdatastoreListVC[i], preferredDatastorePaths, true, i)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			preferredDatastorePathsNew = append(preferredDatastorePathsNew, paths...)
+			pathsLen := len(paths)
+			for j := 0; j < pathsLen; j++ {
+				index := i + j
+				if val, ok := ClusterdatastoreListVC[index][paths[j]]; ok {
+					dsUrls = append(dsUrls, val)
+				}
+			}
+		}
+		preferredDatastorePaths = append(preferredDatastorePaths, preferredDatastorePathsNew...)
+		defer func() {
+			ginkgo.By("Remove preferred datastore tag")
+			for i := 0; i < len(preferredDatastorePathsNew); i++ {
+				err = detachTagCreatedOnPreferredDatastore(masterIp, sshClientConfig, preferredDatastorePathsNew[i],
+					allowedTopologies[0].Values[i], true, i)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Perform scaleup/scaledown operation on statefulsets and " +
+			"verify pv affinity and pod affinity")
+		for i := 0; i < len(statefulSets); i++ {
+			performScalingOnStatefulSetAndVerifyPvNodeAffinity(ctx, client, scaleUpReplicaCount,
```

Comment:
Same as above comment

Reply:
In this testcase, We are creating 3 StatefulSets and both the util takes 1 StatefulSet at a time. Hence, used for loop here